sklearn/svm/classes.py

```diff
@@ -366,6 +366,9 @@ def fit(self, X, y):
         if self.C < 0:
             raise ValueError("Penalty term must be positive; got (C=%r)"
                              % self.C)
+        if ((sample_weight is not None) and
+            np.atleast_1d(sample_weight).ndim > 1):
+                sample_weight = column_or_1d(sample_weight, warn=True)
```

Comment:
I don't think we want to support column vectors. A 1d array should be what we expect. I think columns are accepted elsewhere for historical reasons.

Validation for `sample_weight` is otherwise performed in `_fit_liblinear`